Source/WebCore/Modules/cookie-store/CookieStore.cpp

```diff
@@ -210,79 +241,111 @@ void CookieStore::set(CookieInit&& options, Ref<DeferredPromise>&& promise)
         return;
     }
 
-    if (!m_cookieJar) {
-        promise->reject(SecurityError);
-        return;
-    }
-
-    auto& document = *downcast<Document>(context);
-    // The maximum attribute value size is specified at https://wicg.github.io/cookie-store/#cookie-maximum-attribute-value-size.
-    static constexpr auto maximumAttributeValueSize = 1024;
+    ALWAYS_LOG_WITH_STREAM(stream << "rupin log - point 1");
 
-    Cookie cookie;
-    cookie.name = WTFMove(options.name);
-    cookie.value = WTFMove(options.value);
-    cookie.created = WallTime::now().secondsSinceEpoch().milliseconds();
+    auto setCookies = [options = WTFMove(options), promise = WTFMove(promise)](ScriptExecutionContext& context) {
```

Comment:
`options = crossThreadCopy(WTFMove(options))` now that multiple threads are involved.